core/src/main/scala/gnieh/diffson/package.scala

```diff
@@ -15,6 +15,43 @@
 */
 package gnieh
 
+import scala.collection.immutable.Queue
+
 /** This package contains an implementation of Json JsonPatch, according to [RFC-6902](http://tools.ietf.org/html/rfc6902)
  */
-package object diffson
+package object diffson {
+
+  private val IsDigit = "(0|[1-9][0-9]*)".r
+
+  type Pointer = Queue[Either[String, Int]]
+
+  implicit class PointerOps(val pointer: Pointer) extends AnyVal {
```

Comment:
I did some optimization, it is better now, although I still think it is not perfect... There are still two different types `Pointer` and `JsonPointer`, which is not intuitive - how they differ from the first glance. It would be better to leave just one and put all operations there, making `Pointer` a class with `def evaluate(value: JsValue)` public method - but I realized it is harder that I thought, because of cake-patterns...